### PR TITLE
Add some JitBuilder tests based on existing examples

### PR DESCRIPTION
**Pull request by Leonardo2718**
This is some initial work to address issue #897.

In this change, the `ControlFlowTest` is removed from the examples and replaced by an equivalent test case. A test equivalent to the `WorklistTest` example is also added but the example is not removed as there is no other example that demonstrates the use of JitBuilder worklists.

### CODE REVIEW

File: fvtest/jitbuildertest/WorklistTest.cpp [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ ******************************************************************************/
+
+#include "JBTestUtil.hpp"
+
+#include "ilgen/BytecodeBuilder.hpp"
+#include "ilgen/VirtualMachineState.hpp"
+
+#define EXPECT_BICEQ(expected_index, actual_index) do {\
```
Comment (mstoodle) [inline fvtest/jitbuildertest/WorklistTest.cpp]: why BIC ? I was expecting BCI, but maybe I'm misreading your intent?

File: fvtest/jitbuildertest/WorklistTest.cpp [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+   EXPECT_EQ((expected_index), (actual_index)) << "Unexpected bytecode index (likely implies bytecode was not found in worklist)."; \
```
Comment (mstoodle) [inline fvtest/jitbuildertest/WorklistTest.cpp]: I think the indices not being equal implies the worklist either missed or added a bytecode it should not have. "not found in worklist" implies some kind of search which I think is misleading in this use case.

File: fvtest/jitbuildertest/WorklistTest.cpp (hunk 1 above)
Comment (Leonardo2718) [inline fvtest/jitbuildertest/WorklistTest.cpp]: :stuck_out_tongue: Definitely should be BCI. Fixing.

File: fvtest/jitbuildertest/WorklistTest.cpp (hunk 2 above)
Comment (Leonardo2718) [inline fvtest/jitbuildertest/WorklistTest.cpp]: Good point. Fixed.

**Comment by Leonardo2718** [comment]
@mstoodle Have you had a chance to look over my changes?

File: fvtest/jitbuildertest/WorklistTest.cpp [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ ******************************************************************************/
+
+#include "JBTestUtil.hpp"
+
+#include "ilgen/BytecodeBuilder.hpp"
+#include "ilgen/VirtualMachineState.hpp"
+
+#define EXPECT_BCIEQ(expected_index, actual_index) do {\
+   EXPECT_EQ((expected_index), (actual_index)) << "Unexpected bytecode index (implies bytecode was either missed or erroniously added in worklist)."; \
```
Comment (mstoodle) [inline fvtest/jitbuildertest/WorklistTest.cpp]: typo: should be erroneous

**Comment by mstoodle** [comment]
sorry for the late review, @Leonardo2718 . One typo and I'll merge. Please feel free to poke me faster if I don't respond, but I'll try to pay more attention this time :) .

File: fvtest/jitbuildertest/WorklistTest.cpp (hunk 3 above)
Comment (Leonardo2718) [inline fvtest/jitbuildertest/WorklistTest.cpp]: Fixed.

**Pull request merged**